app/src/main/java/com/expidev/gcmapp/MainActivity.java

```diff
@@ -589,13 +601,33 @@ private void addChurchMarkersToMap() {
                         break;
                 }
 
-                if (render) {
+                if (render && canViewChurch(church.getSecurity())) {
                     clusterManager.addItem(new ChurchMarker(church));
                 }
             }
         }
     }
 
+    private boolean canViewChurch(int churchSecurity)
+    {
+        if(mAssignment == null)
+        {
+            return false;
+        }
+
+        switch(churchSecurity)
+        {
+            case 0:
+                return mAssignment.isLeader() || mAssignment.isMember();
+            case 1:
+                return mAssignment.isLeadership();
+            case 2:
```

Comment:
there is a church security of 3 from legacy data that should be treated as 2.

Long term church security will be converted to an enum which will capture this legacy data value.